org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/wizard/template/WorkspaceFileGenerator.java

```diff
@@ -49,12 +54,15 @@ protected void execute(IProgressMonitor monitor) throws CoreException, Invocatio
 			IWorkspace workspace = ResourcesPlugin.getWorkspace();
 			for (Map.Entry<String, CharSequence> fileEntry : files.entrySet()) {
 				IFile file = workspace.getRoot().getFile(new Path(fileEntry.getKey()));
-				file.create(new StringInputStream(fileEntry.getValue().toString()), true, subMonitor);
+				file.create(new StringInputStream(fileEntry.getValue().toString(), file.getCharset()), true, subMonitor);
 				if (firstFile == null) {
 					firstFile = file;
 				}
 				subMonitor.worked(1);
 			}
+		} catch (UnsupportedEncodingException e) {
```

Comment:
as the api is still beta, should we change it to throws Exception @ArneDeutsch ?
then we would not need these

Reply:
I wouldn't do that. As a user I can't set the encoding so what should I do if the encoding is not supported. Throwing a Core exception should be enough.

Reply:
well. this is about geneneral exception handling.
throws CoreException, InvocationTargetException, InterruptedException {
=> replace this with "throws Exception" as we dont care about the other exceptons neither

Reply:
No because then please only throw RuntimeExceptions :-)
Seriously, I would encapsulate the incovationTargetException and InterruptedException as well in an core Exception and only throw this, because this is handled by the eclipse frame work (with an error dialog if I remember correctly).
Enforcing a lot of catches just blows the code using the method and I mostly just throw a runtimeException in this cases.

Reply:
Another thought: Why would I need 3 different exception, which tell me: "the generation failed!" I have to deal with that, I don't want to catch different types of reasons. If I relly need to know them, e.getReason() is the method to go. But usually I need to do some clean-up or other stuff.
More important than different exception is a good javadoc, which is still missing here.

Reply:
in this case we have explicit handing for some at caller place

Reply:
The `#execute` method is inherited from `WorkspaceModifyOperation` - we cannot add a checked exception to the signature.
I think wrapping it in a CoreException is the right move + some JavaDoc on the failure scenarios.